index.js

```diff
@@ -49,6 +65,8 @@ function copy(src) {
 			dst = { message: src.message };
 		} else if (isBoolean(src) || isNumber(src) || isString(src)) {
 			dst = Object(src);
+		} else if (isTypedArray(src)) {
+			dst = src.constructor.from(src);
```

Comment:
`.constructor` is not reliable/robust - we'd want to use https://npmjs.com/which-typed-array instead (this can be used as an `isTypedArray` check as well, instead of pulling in `is-typed-array` directly)

Reply:
This does not seem to correctly handle node `Buffer`s which is what I was hoping to also cover 🤔

Using the following code:
```js
function isTypedArray(obj) { return !!whichTypedArray(obj); }
// ---
dst = global[whichTypedArray(src)].from(src);
```

I ran the following test:
```js
const buffer = Buffer.from('hi there');
const cloned = traverse.clone(buffer);

console.info({ buffer, cloned });
// {
//   buffer: <Buffer 68 69 20 74 68 65 72 65>,
//   cloned: Uint8Array(8) [ 104, 105, 32, 116, 104, 101, 114, 101 ]
// }
```

Its vastly better than the crash from before though! Any suggestions on how this could be achieved?

Reply:
I have changed this, will need to work buffers out later I think!